src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -84,43 +92,123 @@ private Task OnProjectFactoryCompletedAsync()
+                    return;
+                }
+            }
+
+            var newProjectContext = await _projectContextProvider.UpdateProjectContextAsync(targetFrameworks, CancellationToken.None).ConfigureAwait(false);
+
+            AggregateWorkspaceProjectContext previousProjectContext;
+            lock (_projectContextProvider)
+            {
+                if (newProjectContext == _projectContext)
+                {
+                    // Another thread has already completed the update.
+                    Requires.Range(string.Equals(targetFrameworks, _latestTargetFrameworks, StringComparison.OrdinalIgnoreCase), nameof(targetFrameworks));
+                    return;
+                }
+
+                previousProjectContext = _projectContext;
+                _projectContext = newProjectContext;
+                _latestTargetFrameworks = targetFrameworks;
+            }
+
+            await ResetSubscriptionsAsync().ConfigureAwait(false);
+
+            foreach (var innerContext in previousProjectContext?.InnerProjectContexts)
+            {
+                foreach (var handler in Handlers)
+                {
+                    await handler.Value.OnContextReleasedAsync(innerContext).ConfigureAwait(false);
+                }
+            }
+        }
+
+        private async Task ResetSubscriptionsAsync()
+        {
+            // TODO: Design time builds for all active configured projects (https://github.com/dotnet/roslyn-project-system/issues/532)
+            //_designTimeBuildSubscriptionLink?.Dispose();
+            _evaluationSubscriptionLink?.Dispose();
+
+            using (_tasksService.LoadedProject())
+            {
+                var currentProjects = await _targetFrameworkConfiguredProjectProvider.GetConfiguredProjectsAsync().ConfigureAwait(false);
+
+                // TODO: Design time builds for all active configured projects (https://github.com/dotnet/roslyn-project-system/issues/532)
+                //var sourceBlocks = currentProjects.Select(
+                //    cp => cp.Services.ProjectSubscription.JointRuleSource.SourceBlock.SyncLinkOptions<IProjectValueVersions>());
+                //var target = new ActionBlock<Tuple<ImmutableList<IProjectValueVersions>, TIdentityDictionary>>(s => ProjectPropertyChangedAsync(s, RuleHandlerType.DesignTimeBuild));
+                //_designTimeBuildSubscriptionLink = ProjectDataSources.SyncLinkTo(sourceBlocks.ToImmutableList(), target, null);
+
+                var sourceBlocks = currentProjects.Select(
+                    cp => cp.Services.ProjectSubscription.ProjectRuleSource.SourceBlock.SyncLinkOptions<IProjectValueVersions>());
+                var target = new ActionBlock<Tuple<ImmutableList<IProjectValueVersions>, TIdentityDictionary>>(s => ProjectPropertyChangedAsync(s, RuleHandlerType.Evaluation));
+                _evaluationSubscriptionLink = ProjectDataSources.SyncLinkTo(sourceBlocks.ToImmutableList(), target, null);
```

Comment:
@natidea @davkean - this is how I am trying to enable evaluations for all active configured projects. `ProjectPropertyChangedAsync` seems to be called only for single active configured project (with the first TargetFramework).

I have the following in my project file:
```
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFrameworks>netcoreapp1.0;netcoreapp1.1</TargetFrameworks>
  </PropertyGroup>

  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp1.0'">
    <Compile Include="Program.cs" />
  </ItemGroup>
  <ItemGroup Condition="'$(TargetFramework)' != 'netcoreapp1.0'">
    <Compile Include="Program.cs" />
  </ItemGroup>
```

Changes to the first item group causes `ProjectPropertyChangedAsync` to fire, but changes to the second item group doesn't cause it to fire.